text/0155-external-modules.md

```diff
@@ -0,0 +1,124 @@
+both provided by core maintainers of the project and any third party.
+
+In order for a component to be included into Registry its authors have to fill a [self-assessment form](#registry-self-assessment-form).
+
+The registry should allow a clear visibility of component's ownership, quality and compatibility with other Otel components.
+
+A component can be removed from the Registry if any declaration from the self-assessment form is violated and not remedied
+in a timely manner, provisionally one or two weeks.
+
+## Internal details
+
+We distinguish the following sources of OpenTelemetry components and integrations.
+
+### Native or built-in instrumentations
+
+Any library or framework can use the OpenTelemetry API to natively produce telemetry.
+We encourage library authors to submit their library for inclusion into the OpenTelemetry Registry.
+
+### Core components
+
+Otel SIGs may provide instrumentation for any libraries that are deemed important enough by the SIG’s maintainers.
+By doing this the SIG maintainers commit to support this instrumentation (including future versions of the library),
+provide updates/fixes, including security patches and guarantee that they comply with Otel semantic conventions and best practices.
+
+Depending on the SIG, these instances of core instrumentation may share the repository, infrastructure and maintainers with
+the OpenTelemetry API/SDK implementation for this language or be separate.
+
+All instances of core instrumentation must be included into the Otel Registry following the usual process described above.
+
+### Contrib components
+
+Any language SIG may have one or more “contrib” repos containing components contributed by external developers.
+These repositories may have a separate set of approvers/maintainers than the core API/SDK repo.
+SIG maintainers are encouraged to have a simplified process to promote contributors to approver/maintainer role in this repository.
+A contrib repository may leverage the CODEOWNERS functionality of GitHub to assign maintainers to individual packages
+even if this means granting write permissions to the whole repo.
+The goal should be to distribute the load of reviewing PRs and accepting changes as much as possible.
+
+All components in a contrib repository are expected to be included into the Otel Registry following the usual process described above.
+
+It should be very easy for external contributors to include their components into a contrib repo as opposed to hosting them separately.
+This way they can reuse existing infrastructure for testing, publishing, security scanning etc.
+This will also greatly simplify responsibility transfer between different maintainers if their priorities change. It also promotes the development
+and maintenance of a single instrumentation package for each instrumentation source, so that work isn't spread amongst multiple parallel solutions.
```

Comment:
So if two developers came up with the different models to instrument a library, and do not want to collaborate. Or even worse, have slightly different semantics of attributes. Do we include both? Who decides on which wins?

Reply:
Most probably one of the integrations will be the first and already included into contrib repo. Then we first ask the second developer to try and collaborate with the first one. If this fails, then the second developer is free to host their vision in their own repo. Both integrations may be included into the Registry.

Not an ideal situation, but I don't have any better answer.

Reply:
>  then the second developer is free to host their vision in their own repo. Both integrations may be included into the Registry.

I have seen this in the past. Indeed, there's no perfect solution, we can only mention our advise for such case.